R/utils.R

```diff
@@ -1,125 +1,152 @@
-drop_null <- function(x){
+# drop_null --------------------------------------------------------------------
+drop_null <- function(x) {
+
   x[lengths(x) != 0]
 }
 
+# get_root_path ----------------------------------------------------------------
+get_root_path <- function(pkg) {
 
-#' @importFrom utils installed.packages
-get_root_path <- function(pkg){
-  installed <- installed.packages()
-  if(pkg %in% installed[,1]){
-    root <- base::system.file(".", package = pkg)
-  } else if(is_package(file.path(pkg))){
-    root <- pkg
-  } else {
-    if(is.character(pkg)){
-      root <- pkg # use pkg as guess anyway
-#    } else {
-#      root <- "." # stick with default
-    }
+  if (pkg %in% installed_package_names()) {
+
+    package_file(pkg, ".")
+
+  } else if (is_package(file.path(pkg))) {
+
+    pkg
+
+  } else if (is.character(pkg)) {
+
+    pkg # use pkg as guess anyway
+
+# } else {
+#
+#   "." # stick with default
   }
-  root
 }
 
+# package_file -----------------------------------------------------------------
+# Just a shortcut to system.file(..., package = pkg)
+package_file <- function(pkg, ...) {
 
-## Like system.file, but pkg can instead be path to package root directory
-get_file <- function(FILE, pkg = "."){
-  f <- file.path(pkg, FILE)
-  if(file.exists(f))
-    f
-  else {
-    f <- system.file(FILE, package = pkg)
-  }
+  base::system.file(..., package = pkg)
```

Comment:
why prefixing with `base::`?

Reply:
I kept it as it was before but also thought that it can be omitted. I will remove "base::"